### PR TITLE
Fixes #2888 Marketing Cloud Event Specificity

### LINKED ISSUE #2888: Marketing Cloud Event Specificity
## Problem/Motivation
Marketing Cloud event dates don't specify the iteration of an event if repeating.

### Describe the bug
PHP is not implemented like in `az_event`.

### To Reproduce
1. Add repeating event to a site
2. Open the Marketing Cloud Content for Email page
3. Open 2 different dates of events
4. View the same date displayed in the rendered view

### Module updates
Update the `.module` file to have the same functionality as `az_event` for `az_event_month` and `az_event_day`

### View updates - Complete
1. Change path of Event Layout display to `export/marketing-cloud/event-layout/%node/%field_az_event_date_value_1`
2. Add date field to Admin display - HTML Datetime format, exclude from display
3. Change Event link to `<a href="/export/marketing-cloud/event-layout/{{ nid }}/{{ field_az_event_date_value_1 }}">Event Layout</a>`

### PR DESCRIPTION
**Pull request by tadean**
Some followup commits added to @danahertzberg 's branch:

Currently, display of marketing cloud events are not aware of their recurrence relationship, meaning that when you create the marketing cloud output for a view, it is always displayed as the first occurrence in the series, even if the item they copied was a later event in the series.

## Description
This happens because the view in question is not delta-aware. Events are preprocessed by `az_event` to modify the field values if a valid views delta relationship is detected. The calendar views pass along the delta relationship, but the `az_marketing_cloud` views do not.

This PR adds the delta relationship to the event display of the marketing cloud view.

## Related issues
Closes #2888 

## How to test

- enable `az_marketing_cloud`
- enable `az_demo`
- proceed to `/admin/content/marketing-cloud`
- copy the **Event Layout** url for an event (make sure it is not a news item)
- verify the rendered date matches the actual occurrence of the event you selected

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [x] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tadean** [comment]
- @danahertzberg 's original commit used `ISO 8601` as the date format. but I wasn't able to get this working with the view contextual filter. It seems to only work with Y-m-d, even if another format is provided. This might be a contextual filter issue to investigate
- Alternatively, we could use the delta index for the contextual filter rather than a date

**Comment by joeparsons** [comment]
## 2023-11-03 Friday Meeting Notes
@tadean Wants to investigate why different date format strings don't work as contextual filters before finalizing/merging this.

**Review by danahertzberg — DISMISSED** [review]

**Comment by danahertzberg** [comment]
Determined that URL should contain `/{{ nid }}/instance/{{ delta }}`.
@tadean Will take this from here.

**Comment by tadean** [comment]
Resolved merge conflict and updated approach to use a delta-based index for the path of the event display.

**Review by danahertzberg — APPROVED** [review]

File: modules/custom/az_marketing_cloud/config/install/views.view.az_marketing_cloud.yml
Comment (bberndt-uaz) [inline modules/custom/az_marketing_cloud/config/install/views.view.az_marketing_cloud.yml:1089]: More background on this issue with `format_plural_string`: https://www.drupal.org/project/drupal/issues/2829919.

I don't believe this issue needs to block this PR, especially since `format_plural` is false in both cases in this file (as @danahertzberg noted).

**Review by bberndt-uaz — APPROVED** [review]

File: modules/custom/az_marketing_cloud/config/install/views.view.az_marketing_cloud.yml
Comment (tadean) [reply modules/custom/az_marketing_cloud/config/install/views.view.az_marketing_cloud.yml:1089]: I agree - it's a bit of a symfony encoding oddity but doesn't seem to have any actual effect. Thanks @bberndt-uaz !

**Review by joeparsons — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**
